### PR TITLE
Add _typeshed.sentinel

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
Closes: #10480, alternative to #10576

File: stdlib/argparse.pyi [hunk 1]
```diff
@@ -334,7 +335,21 @@ class Action(_AttributeHolder):
...
+        def __init__(
+            self,
+            option_strings: Sequence[str],
+            dest: str,
+            default: _T | str | None = None,
+            type: Callable[[str], _T] | FileType | None = sentinel,  # noqa: Y011
+            choices: Iterable[_T] | None = sentinel,  # noqa: Y011
+            required: bool = False,
+            help: str | None = None,
+            metavar: str | tuple[str, ...] | None = sentinel,  # noqa: Y011
```
Comment (AlexWaygood) [inline stdlib/argparse.pyi:349]: I suppose you'll be wanting to add special-casing for this in flake8-pyi? :)

Comment (srittau) [reply stdlib/argparse.pyi:349]: Indeed.

File: tests/stubtest_allowlists/py312.txt [hunk 2]
```diff
@@ -1,5 +1,4 @@
 # Uncategorised, from Python 3.12
-argparse.BooleanOptionalAction.__init__
```
Comment (srittau) [inline tests/stubtest_allowlists/py312.txt:2]: This is a clear gain, in my opinion. Previously, all stubtest checks for this method were disabled, now it just works.

**Review by AlexWaygood — APPROVED** [review]
This looks like a pretty nice solution, thanks. I confirmed locally that mypy still errors out if you do something like `BooleanOptionalAction(['foo'], 'bar', 'baz', type=12345)` (i.e., mypy just ignores the default value when deciding which values are legal to be passed to the call; it only looks at the type annotation).

I tested using `mypy -c "from argparse import BooleanOptionalAction; b = BooleanOptionalAction(['foo'], 'bar', 'baz', type=12345)" --custom-typeshed-dir .`

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**